### PR TITLE
Feat/av 25/flight controller

### PR DESCRIPTION
**Pull request by Klonotoros**

### CODE REVIEW

File: internal/controller/contact.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func (c *contactController) UpdateContact(ctx *gin.Context) {
 // @Failure 400 {object}      util.HTTPError
 // @Failure 404 {object}      util.HTTPError
 // @Failure 500 {object}      util.HTTPError
-// @Router  /api/contacts/{id} [delete]
```
Comment (matisiekpl) [inline internal/controller/contact.go:151]: Why removed `api` prefix?

File: internal/controller/logbook.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+//
+// @Summary Get user logbook entries
+// @Description Get a list of logbook entries for a user
+// @Tags logbook
+// @Produce  json
+// @Security ApiKeyAuth
+// @Success 200 {object} []dto.LogbookResponse
+// @Failure 500 {object}      util.HTTPError
+// @Failure 400 {object}      util.HTTPError
+// @Router  /logbook [get]
```
Comment (matisiekpl) [inline internal/controller/logbook.go:40]: Are you sure of this path?

File: internal/controller/logbook.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	if err := ctx.ShouldBindJSON(&getLogbookRequest); err != nil {
+		util.NewError(ctx, http.StatusBadRequest, err)
+		return
+	}
+
+	if getLogbookRequest.Start == nil && getLogbookRequest.End != nil || getLogbookRequest.Start != nil && getLogbookRequest.End == nil {
+		util.NewError(ctx, http.StatusBadRequest, errors.New("both start and end time must be provided or neither"))
+		return
+	} else if getLogbookRequest.Start == nil && getLogbookRequest.End == nil {
+		start = time.Now()
```
Comment (matisiekpl) [inline internal/controller/logbook.go]: Invalid. `end` should be `time.Now()`

File: internal/controller/logbook.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+		end = start.AddDate(0, -3, 0)
```
Comment (matisiekpl) [inline internal/controller/logbook.go]: And here `start` should be -90 days

File: internal/controller/logbook.go [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+// @Failure 500 {object}      util.HTTPError
+// @Router /logbook/{id} [put]
+func (c *logbookController) UpdateLogbookEntry(ctx *gin.Context) {
+	flightID, err := strconv.ParseUint(ctx.Param("id"), 10, 32)
+	if err != nil {
+		util.NewError(ctx, http.StatusBadRequest, err)
+		return
+	}
+
+	userID := ctx.GetString("userID")
```
Comment (matisiekpl) [inline internal/controller/logbook.go]: Introduce a `const string` for this key?

File: internal/service/logbook.go [hunk 6]
```diff
@@ -210,45 +213,52 @@ func (l *logbookService) InsertLogbookEntry(userID string, logbookRequest dto.Lo
...
-		return errors.New("flight does not belong to user")
+		return fmt.Errorf("%w: %v", dto.ErrBadRequest, "flight does not belong to user")
 	}
 
 	tx := l.flightRepository.Begin()
 
 	if err := l.landingRepository.DeleteByFlightIDTx(tx, flightID); err != nil {
 		tx.Rollback()
-		return err
+		return err // internal
```
Comment (matisiekpl) [inline internal/service/logbook.go]: Remove comment

**Review by matisiekpl — CHANGES_REQUESTED** [review]

File: internal/controller/contact.go (hunk 1 above)
Comment (Klonotoros) [reply internal/controller/contact.go:151]: Because base path is documented in main.go

File: internal/controller/logbook.go (hunk 2 above)
Comment (Klonotoros) [reply internal/controller/logbook.go:40]: yes, whats wrong?

**Review by matisiekpl — APPROVED** [review]

**Pull request merged**